src/model/immutable/__tests__/ContentState-test.js

```diff
@@ -12,156 +12,130 @@
-      var singleBlock = getSample(SINGLE_BLOCK);
-      var key = SINGLE_BLOCK[0].key;
-      expect(singleBlock.getKeyBefore(key)).toBe(undefined);
-      expect(singleBlock.getKeyAfter(key)).toBe(undefined);
-
-      var multiBlock = getSample(MULTI_BLOCK);
-      var firstKey = MULTI_BLOCK[0].key;
-      var secondKey = MULTI_BLOCK[1].key;
-
-      expect(multiBlock.getKeyBefore(firstKey)).toBe(undefined);
-      expect(multiBlock.getKeyAfter(firstKey)).toBe(secondKey);
-      expect(multiBlock.getKeyBefore(secondKey)).toBe(firstKey);
-      expect(multiBlock.getKeyAfter(secondKey)).toBe(undefined);
-    });
+const getSample = textBlocks => {
+  const contentBlocks = textBlocks.map(block => new ContentBlock(block));
+  const blockMap = BlockMapBuilder.createFromArray(contentBlocks);
+  return new ContentState({
+    blockMap,
+    selectionBefore: new SelectionState(),
+    selectionAfter: new SelectionState(),
   });
+};
+
+beforeEach(() => {
+  contentState = ContentState.createFromText('');
+  jest.resetModules();
+});
+
+test('must create a new instance', () => {
+  const state = getSample(SINGLE_BLOCK);
+  expect(state instanceof ContentState).toMatchSnapshot();
+});
+
+test('must create properly with an empty block array', () => {
+  const state = ContentState.createFromBlockArray([]);
+  expect(state instanceof ContentState).toMatchSnapshot();
+});
 
-  describe('block fetching', () => {
-    it('must retrieve or fail fetching block for key', () => {
-      var state = getSample(SINGLE_BLOCK);
-      var block = state.getBlockForKey('a');
-      expect(block instanceof ContentBlock).toBe(true);
-      expect(block.getText()).toBe(SINGLE_BLOCK[0].text);
-      expect(state.getBlockForKey('x')).toBe(undefined);
-    });
+test('key fetching must succeed or fail properly', () => {
+  const singleBlock = getSample(SINGLE_BLOCK);
+  const key = SINGLE_BLOCK[0].key;
+  const multiBlock = getSample(MULTI_BLOCK);
+  const firstKey = MULTI_BLOCK[0].key;
+  const secondKey = MULTI_BLOCK[1].key;
+
+  expect(singleBlock.getKeyAfter(key)).toMatchSnapshot();
+  expect(singleBlock.getKeyBefore(key)).toMatchSnapshot();
+  expect(singleBlock.getKeyAfter(key)).toMatchSnapshot();
+
+  expect(multiBlock.getKeyBefore(firstKey)).toMatchSnapshot();
+  expect(multiBlock.getKeyAfter(firstKey)).toMatchSnapshot();
+  expect(multiBlock.getKeyBefore(secondKey)).toMatchSnapshot();
+  expect(multiBlock.getKeyAfter(secondKey)).toMatchSnapshot();
+});
+
+test('block fetching must retrieve or fail fetching block for key', () => {
+  const state = getSample(SINGLE_BLOCK);
+  const block = state.getBlockForKey('a');
+
+  expect(block instanceof ContentBlock).toMatchSnapshot();
+  expect(block.getText()).toMatchSnapshot();
+  expect(state.getBlockForKey('x')).toMatchSnapshot();
+});
+
+test('must create entities instances', () => {
+  const contentState = createLink();
+  expect(typeof contentState.getLastCreatedEntityKey()).toMatchSnapshot();
```

Comment:
Nit - since I know you didn't chjange the logic here;
I wonder why it checks `typeof`? I am guessing because the entityKey is different every time, so we can only check that it was created, not that it's correct.

Reply:
That is correct